### PR TITLE
refactor(__alpineDevtool in backend.js to a class)

### PR DESCRIPTION
**Pull request by HugoDF**
- refactor to a class + add init() function to get rid of `window.__alpineDevtools` shenanigans

### CODE REVIEW

File: packages/shell-chrome/src/backend.js [hunk 1]
```diff
@@ -121,113 +26,230 @@ function serializeDataProperty(value) {
...
-    const allComponentsInitialized = Object.values(rootEls).every((e) => e.__alpineDevtool)
-    if (window.__alpineDevtool.components.length === rootEls.length && allComponentsInitialized) {
-        return false
+class AlpineDevtoolsBackend {
+    constructor() {
+        this.components = []
+        this.uuid = 1
+        this.hoverElement = null
+        this.observer = null
+        this._stopMutationObserver = false
```
Comment (ryangjchandler) [inline packages/shell-chrome/src/backend.js:35]: I assume we're using `_` prefix here to indicate a private field?

Comment (HugoDF) [reply packages/shell-chrome/src/backend.js:35]: yep, the mutation stuff is now all handled inside of the class

**Comment by HugoDF** [comment]
Note that we could push this further and move the "init" function into the class but I reckon we'll lose some flexibility with that, it's cool to keep the stateful code in a class but the message handling is very procedural (and not that stateful)

**Comment by ryangjchandler** [comment]
> Note that we could push this further and move the "init" function into the class but I reckon we'll lose some flexibility with that, it's cool to keep the stateful code in a class but the message handling is very procedural (and not that stateful)

Yeah, keep the class stateful. If it didn't have nested functions, I'd probably say make it a static function that we can just call.

**Comment by HugoDF** [comment]
> > Note that we could push this further and move the "init" function into the class but I reckon we'll lose some flexibility with that, it's cool to keep the stateful code in a class but the message handling is very procedural (and not that stateful)
> 
> Yeah, keep the class stateful. If it didn't have nested functions, I'd probably say make it a static function that we can just call.

anything that isn't obvious in the code? I think being able to use `this` instead of `window.__alpineDevtool` is a win in of itself 😄 but would be good to get your thoughts

**Comment by HugoDF** [comment]
ok merging this, hopefully doesn't mess with anyone's work in progress apart from @ryangjchandler's "dispatch interception" one (#129)

**Pull request merged**